pkg/predicate/predicate.go

```diff
@@ -48,6 +48,8 @@ var _ Predicate = Funcs{}
 var _ Predicate = ResourceVersionChangedPredicate{}
 var _ Predicate = GenerationChangedPredicate{}
 var _ Predicate = AnnotationChangedPredicate{}
+var _ Predicate = NamespacePredicate{}
```

Comment:
You can achieve the same with 

```
NewPredicateFuncs(func(o client.Object) bool {
   return o.GetNamespace() == "my-namespace" 
}
```

Rather than adding so many predicates I think it would be preferrable to encourage ppl to write their own.

Reply:
is `NewPredicateFuncs` not well documented or hard to find or what prompted you to implement this?

Reply:
Oh， I come from `kubebuilder`. This method is not mentioned in `kubebuilder` docs, so I may miss it.